### ISSUE TITLE
Failure to find temp space on Android

### ISSUE THREAD

**Issue by GoogleCodeExporter** [google-code-export/mythbox#224]
```
What steps will reproduce the problem?
1.Install XBMC for Android
2.Install MythBox
3.Launch MythBox

What is the expected output? What do you see instead?
I was hoping to see MythBox launch. Instead I saw a cut-off error about "No 
usable temp directory found in..." My assumption is that the plug-in simply 
hasn't been tested and/or updated for Android, but I'd be happy to help out and 
test as is possible.

What version of the product are you using? On what operating system?
I was using yesterday's XMBC for Android nightly on an Jelly Bean (4.1) IPTV 
stick (RikoMagic MK802 III). Also reproduced the issue on a Galaxy Nexus 
running 4.1.

Please provide any additional information below.
MythBox would by far be the best MythTV front-end solution for the greatly 
increasing number of Android IPTV systems. I'd really like to help make it 
work. 

```

Original issue reported on code.google.com by `cla...@gmail.com` on 30 Nov 2012 at 4:18

**Comment by GoogleCodeExporter** [comment]
```
same issue...
```

Original comment by `st.thom...@gmail.com` on 6 Dec 2012 at 4:57

**Comment by GoogleCodeExporter** [comment]
```
I too have this issue.
I have XBMC running on the G-Box Midnight v2.1, running Android 4.0.
Sorry to say, I could not find any log files, but then again I do not know 
where to look.

From what I have seen, I much prefer to keep using MythBox versus the LiveTV 
interface built into "Frodo". It is a much cleaner and more "eye catching" 
interface.

... thanks ...
... Dave ...
```

Original comment by `dwarner2...@gmail.com` on 7 Dec 2012 at 12:13

**Comment by GoogleCodeExporter** [comment]
```
Yea getting [err no 2] on a rikomagic ug802 running 4.0.4 rooted, with the beta 
2, and on a stock nexus 7 running 4.2.1 with the beta 3
```

Original comment by `mrrog...@gmail.com` on 9 Dec 2012 at 2:22

**Comment by GoogleCodeExporter** [comment]
```
It appears an Android platform needs specified in platform.py.
```

Original comment by `cla...@gmail.com` on 15 Dec 2012 at 3:35

**Comment by GoogleCodeExporter** [comment]
```
I have a prototype working currently. Well, I should say that it gets past the 
error but I don't currently have a backend to connect to. Luckily, my friend 
Clark does. I will be giving him a copy today for him to test.

As for what might be wrong, I know what the root cause is, however clarkt's 
suggestion of defining an Android platform may also be necessary, which is 
where I started so I haven't tried the complete fix without an AndroidPlatform 
class in place. That being said, the root cause is the Twitter library being 
used by mythbox which is using Python's tempfile library. Unfortunately, that 
library doesn't know how to handle Android, as there aren't any of the 
temporary directories one might expect on a regular *nix system. One would 
think just creating these directories would solve the issue (/tmp, /usr/tmp, 
etc.) however I didn't have much luck there and suspect that Android's app 
security measure might be getting in the way?

tl;dr -- the solution is to modify the twitter library to force the temporary 
dir/file placement to occur within the XBMC directories 
(/data/data/org.xbmc.xbmc/cache/temp) by manually changing the tempfile.tempdir 
variable.

As I said, once it's tested, I'll let you know.

-omegamike

```

Original comment by `omegami...@gmail.com` on 19 Dec 2012 at 6:43

**Comment by GoogleCodeExporter** [comment]
```
Also, I forgot to mention, it seems like there might be a bug in the current 
Android build of XBMC that isn't completely removing addons when they are 
uninstalled. It kept reloading an old cached version for me, which was a point 
of frustration, FYI.

-omegamike

```

Original comment by `omegami...@gmail.com` on 19 Dec 2012 at 6:59

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
Did this issue ever get resolved?  I appear to have this problem, having 
installed XBMC on a Nexus last night.
```

Original comment by `itsdomin...@gmail.com` on 30 Mar 2013 at 8:57

**Comment by GoogleCodeExporter** [comment]
```
Also occurring on OUYA build of XBMC 
```

Original comment by `robert.c...@gmail.com` on 20 Sep 2013 at 4:44

**Comment by GoogleCodeExporter** [comment]
```
Did the changed tempfile.py work?
```

Original comment by `jeteb...@gmail.com` on 23 Jul 2014 at 11:01

**Comment by GoogleCodeExporter** [comment]
```
It's now November 2014 and I don't see an update. I see the problem on Nexus 
and now Amazon Firestick under Gotham.
```

Original comment by `k...@mathcs.emory.edu` on 24 Nov 2014 at 2:42

**Comment by GoogleCodeExporter** [comment]
```
Bump
```

Original comment by `ayeuse...@gmail.com` on 1 Jan 2015 at 12:50
